core/src/main/java/org/hisp/dhis/android/sdk/synchronization/domain/enrollment/EnrollmentSynchronizer.java

```diff
@@ -48,6 +53,20 @@ public void sync(Enrollment enrollment) {
         }
     }
 
+    public void sync(List<Enrollment> enrollments) {
+        Collections.sort(enrollments, new Enrollment.EnrollmentComparator());
+
+        for (Enrollment enrollment : enrollments) {
+            if(enrollment.isFromServer()){
```

Comment:
I think that this condition is problematic. If in a synced enrollment I create a new event, I edit or I delete when I sync TEI, this condition avoid that the events with changes being sent. I think we have 2 options:
- When changes is realized in enrollment (add, edit, remove events) change fromServer property to false (I prefer this)
- For TEI with changes send always its enrollments and events